Bottom: 7f355c2c4b9a69f80c172a756e9357d22ad478bf
Top:    6765a0c30671e9913abf7680b7ecc2c0c4415ffc
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-10-24 21:28:37 -0700

Refresh of rs-iomap

---

diff --git a/src/rsocket.c b/src/rsocket.c
index afcc5b1..eb8b3f0 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -2319,7 +2319,7 @@ off_t riomap(int socket, void *buf, size_t len, int prot, int flags, off_t offse
 	int access = IBV_ACCESS_LOCAL_WRITE;
 
 	rs = idm_at(&idm, socket);
-	if ((rs->state != rs_connect_rdwr) || (prot & ~(PROT_WRITE | PROT_NONE)))
+	if (rs->cm_id->pd || (prot & ~(PROT_WRITE | PROT_NONE)))
 		return ERR(EINVAL);
 
 	fastlock_acquire(&rs->iomap_lock);
